### ISSUE TITLE
`plc_fillin` can overflow the input buffer

### ISSUE THREAD

**Issue by MathieuDuponchelle** [freeswitch/spandsp#43]
When calling `plc_fillin` with `plc_state_t *s` and `s->missing_samples == 0`, the function attempts to write `pitch_overlap` bytes to the input buffer, regardless of its length. This causes crashes in a GStreamer application using the `spandsp` plugin, because it might call `plc_fillin` with very small buffers (eg with my test case 2-byte long).

Can the documentation of `plc_fillin` be updated to specify the minimum size of its input buffers, or even better the function be fixed to support any-sized buffers?

Thanks!

**Comment by andywolk** [comment]
Hi @MathieuDuponchelle 
Can you fix that and make a PR?

**Comment by MathieuDuponchelle** [comment]
Hey @andywolk , I wish, but I'm completely unfamiliar with the code base. Perhaps if you guide me through the correct solution I can propose something, but that will probably not be very efficient :)

**Comment by coppice-git** [comment]
That problem should now be fixed. There was a test in the normal receive function that was missing in the fillin version.

**Comment by andywolk** [comment]
@coppice-git shouldn't the file sit in src/ not in root?

**Comment by coppice-git** [comment]
Yeah. I just screwed up, while trying to sort out my access to the repo. I'm trying to revert the change now, and do it properly.

**Comment by MathieuDuponchelle** [comment]
Great, thanks! To be honest though I don't remember how I was testing this so can't confirm the fix works :)
